subprojects/rzwinkd/iob_net.c

```diff
@@ -407,14 +421,14 @@ bool _verifyhmac(iobnet_t *obj) {
 		return false;
 	}
 
-	if (!rz_msg_digest_update(md, obj->buf, obj->size - KDNET_HMAC_SIZE) ||
+	if (!rz_msg_digest_update(md, obj->read_buf, obj->read_size - KDNET_HMAC_SIZE) ||
 		!rz_msg_digest_final(md) ||
 		!(digest = rz_msg_digest_get_result(md, "sha256", NULL))) {
 		rz_msg_digest_free(md);
 		return false;
 	}
 
-	bool result = !memcmp(digest, obj->buf + obj->size - KDNET_HMAC_SIZE, KDNET_HMAC_SIZE);
+	bool result = !memcmp(digest, obj->read_buf + obj->read_size - KDNET_HMAC_SIZE, KDNET_HMAC_SIZE);
```

Comment:
I didn't follow the code here deeply, but please be sure to check the validity of the size for the `memcp()` to avoid OOB

Reply:
added check in https://github.com/rizinorg/rizin/blob/d352906cd3b00080d58199f55dc983f1e4349dd0/subprojects/rzwinkd/iob_net.c#L444